### DISCUSSION TITLE
NVIDIA T400 NVIDIA

### DISCUSSION THREAD

**Discussion by sundeepgoel72** [Breakthrough/DVR-Scan#99] [General]
Will the above card work with current version of dvr-scan on a Linux or windows platform?

Please also advise how to find which card are compatible or not , and how to determine fps for each?

**Comment by Breakthrough** [comment]
The Windows builds I distribute require cards with compute capability ~5.0~  5.2 or higher:

https://en.wikipedia.org/wiki/CUDA#GPUs_supported

The T400 has CC 7.5 so it will work.  Determining actual performance improvement is non-trivial, and there is still a lot of optimization opportunity to be done with respect to CUDA support currently.  [See my comment on #97](https://github.com/Breakthrough/DVR-Scan/issues/97#issuecomment-1221571544) for some benchmarks on my system using a GTX 2070.

I suspect there won't be any significant improvements when running DVR-Scan on more powerful GPUs, at least with the current implementation.  GPU processing should still be faster than CPU processing, but to get anything better will require major changes which likely won't be addressed unless DVR-Scan is rewritten in C++ or Rust.  See #12 for some background on that discussion.

If you manage to compile OpenCV for older cards, please do share some benchmarks here.  And of course, I would be most happy to accept any pull requests for performance enhancements.

**Reply by sundeepgoel72** [reply]
> The Windows builds I distribute require cards with compute capability 5.0 or higher:

GTX 750 Ti is compute 5.0,  I assume that should work - but its not.  please refer #94

**Reply by Breakthrough** [reply]
Sorry I thought it was 5.0, it was actually 5.2 to 8.6.

**Reply by sundeepgoel72** [reply]
> Sorry I thought it was 5.0, it was actually 5.2 to 8.6.

Thanks. One other Q, I was looking at : https://jamesbowley.co.uk/downloads/#OpenCV4.5.0 . It mentions prebuild windows binaries for > 3.5, would these work in my case, or have I misinterpreted the information on that page ?

**Reply by Breakthrough** [reply]
They should work - so long as you can run the following command, you can use the CUDA version:

    python -c "import dvr_scan; assert dvr_scan.motion_detector.MotionDetectorCudaMOG2.is_available()"
